core/src/test/java/org/apache/iceberg/util/TestTruncateUtil.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.util;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import org.assertj.core.api.Assertions;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+class TestTruncateUtil {
+  @Test
+  public void testTruncateByteBuffer() throws Exception {
+    Assert.assertEquals(
+        "Should truncate binary longer than length",
+        ByteBuffer.wrap("abcd".getBytes("UTF-8")),
+        TruncateUtil.truncateByteBuffer(4, ByteBuffer.wrap("abcdefg".getBytes("UTF-8"))));
+    Assert.assertEquals(
+        "Should not pad binary shorter than length",
+        ByteBuffer.wrap("abc".getBytes("UTF-8")),
+        TruncateUtil.truncateByteBuffer(4, ByteBuffer.wrap("abc".getBytes("UTF-8"))));
+  }
+
+  @Test
+  public void testTruncateByte() {
```

Comment:
I don't find these test cases very useful. They're difficult to read because you're using `MAX_VALUE` to pass both width and value, so it isn't clear which one is which (`truncateByte(10, 244)` is much more clear).

I also think there are already tests for most of these cases, so I'd probably remove these and make sure the other paths are covered.

Reply:
Will remove. The pathway is already covered by `TestSparkTruncateFunction`.

Reply:
Removed all of these and left just the test that shows it may or may not throw an exception for invalid input.